File: Data Structures & Algorithms/house-robber/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    rob(nums) {
        const n = nums.length;
        const dp = new Array(n+2).fill(0);
        // 16 15 14 6 6 0 0 
        for(let i = n - 1; i >= 0 ; i--){
            dp[i] = Math.max(dp[i+1], nums[i] + dp[i+2]);
        }

        return dp[0];
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cs
public class Solution {
    public int FindDuplicate(int[] nums) {
        int slow = 0, fast = 0;
        while(true) {
            fast = nums[nums[fast]];
            slow = nums[slow];
            if(slow == fast) break;
        }
        slow = 0;
        while(slow != fast) {
            fast = nums[fast];
            slow = nums[slow];
        }
        return slow;
    }
}
